### ISSUE TITLE
(Webapp) : Minimum rating specified 

### ISSUE THREAD

**Issue by kaustubhhiware** [kaustubhhiware/rose#5]
Once the webapp is completed, and a plot is visualised for any show, we should have a bar / input field that takes an imdb rating for instance, and highlights the episodes with rating higher than the specified value.

Could proceed with displaying the plot first, and then highlighting the bars where value >= specified. Visualization to be discussed before sending PR.

Pointed out by Shalini Mukhopadhyay

**Comment by shaliniiit** [comment]
I am working on this please.

**Comment by kaustubhhiware** [comment]
@shaliniit

**Comment by kaustubhhiware** [comment]
@shaliniiit Thanks for your enthusiasm, however this issue can be worked upon only after #4 is completed, since this is a dependent task. I'm aware you're the one who pointed this out, but as you can see this task depends on a previous issue which is currently being worked upon.

**Comment by shaliniiit** [comment]
Respected sir
As of now even I have not been able to completely grasp the code and get it
running on Anaconda. So I will be very happy to work once #7 gets over as I
need time myself. thanks. Regards.


[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
12/12/18,
10:48:51

**Comment by kaustubhhiware** [comment]
In that case, assigning this issue to you. Will notify you once the previous issue is completed.

**Comment by shaliniiit** [comment]
Respected sir
I would be really obliged if you could let me know if the previous issue
has been resolved.
Thanks

**Comment by kaustubhhiware** [comment]
Yes, so the issue is now currently available since the PR for #4 was merged only today.

We need to do the following :
Have an interactive plot instead of the currently static one. For this, we will modify the scrape_views.py script such that for command line usage, it uses Matplotlib and for webapp, it generates a plotly or d3 plot, which is interactive.

Have a look at #26. Currently a variable named decider is used to distinguish usage between script and webapp calling it, and an image is generated which is hardly the recommended approach.

**Comment by kaustubhhiware** [comment]
@shaliniiit if I was not able to make myself clear, or you have trouble understanding anything, feel free to comment here / mail me. I have intentionally not given a detailed solution straight away, but mentioned only keywords so that you can learn most of the stuff on your own rather than me just handing it out to you.

**Comment by shaliniiit** [comment]
Now if the decider==2 we are using show().
What I must do is generate a d3 plot instead of just printing a 2d image?

**Comment by shaliniiit** [comment]
I am getting
 Internal Server Error
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
whenever I am trying to display Bar chart and Average plot for a TV series without making any changes to the original code.
Is there some special tool I have to install?

**Comment by kaustubhhiware** [comment]
When exactly are you getting the error? Seems to be working fine for me.

**Comment by shaliniiit** [comment]
When I hit submit for a TV series it gives this error

**Comment by shaliniiit** [comment]
For display bar chart and display averaged chart

**Comment by kaustubhhiware** [comment]
Please tick imdb option, and then tick one of averaged or bar chart. Should work.

Since you're developing, instead of running the webapp you should try to execute the flask webapp on your own system using the instructions specified in the README.

**Comment by shaliniiit** [comment]
I have ticked both options and tried to run it on prompt as well on which
it gives Module not found for scrape_views. When I try to install
scrape_views it says version not found. I am trying on another laptop now.
Thanks.

**Comment by shaliniiit** [comment]
It is running now.
But now  there is a connectivity problem in this area.
[image: image.png]


[image: image.png]
.How much time do I have? I think it will be alright in half an hour.
Thanks.



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
26/12/18,
01:06:15

**Comment by shaliniiit** [comment]
In my case the app is showing bar chart and average plot correctly only for Game of Thrones and not for Two and a Half Men.
And running scrape_views.py is giving this error 
![image](https://user-images.githubusercontent.com/32547466/50426145-f20f5900-08ac-11e9-9510-3059fa3d3ba5.png)

**Comment by shaliniiit** [comment]
![image](https://user-images.githubusercontent.com/32547466/50523595-0a72b400-0af7-11e9-9c99-9076fdbfae93.png)
Can you help me with this error please..
